tests/integration/stepfunctions/v2/error_handling/test_task_service_dynamodb.py

```diff
@@ -0,0 +1,117 @@
+import json
+
+import pytest
+
+from localstack.testing.pytest import markers
+from localstack.utils.strings import short_uid
+from tests.integration.stepfunctions.templates.errorhandling.error_handling_templates import (
+    ErrorHandlingTemplate as EHT,
+)
+from tests.integration.stepfunctions.utils import create_and_record_execution, is_old_provider
+
+pytestmark = pytest.mark.skipif(
+    condition=is_old_provider(), reason="Test suite for v2 provider only."
+)
+
+
+@markers.snapshot.skip_snapshot_verify(
+    paths=[
+        "$..loggingConfiguration",
+        "$..tracingConfiguration",
+        "$..previousEventId",
+        # TODO: add support for Sdk Http metadata.
+        "$..SdkHttpMetadata",
+        "$..SdkResponseMetadata",
+        # TODO: review LS's dynamodb error messages.
+        "$..Cause",
+        "$..cause",
+    ]
+)
+class TestTaskServiceDynamoDB:
+    def test_invalid_param(
+        self,
+        aws_client,
+        create_iam_role_for_sfn,
+        create_state_machine,
+        dynamodb_create_table,
+        snapshot,
+    ):
+        snapshot.add_transformer(snapshot.transform.dynamodb_api())
+
+        template = EHT.load_sfn_template(EHT.AWS_SERVICE_DYNAMODB_PUT_ITEM)
+        definition = json.dumps(template)
+
+        exec_input = json.dumps(
+            {"TableName": f"no_such_sfn_test_table_{short_uid()}", "Key": None, "Item": None}
+        )
+        create_and_record_execution(
+            aws_client.stepfunctions,
+            create_iam_role_for_sfn,
+            create_state_machine,
+            snapshot,
+            definition,
+            exec_input,
+        )
+
+    def test_put_item_no_such_table(
+        self,
+        aws_client,
+        create_iam_role_for_sfn,
+        create_state_machine,
+        snapshot,
+    ):
+        snapshot.add_transformer(snapshot.transform.dynamodb_api())
+
+        table_name = f"no_such_sfn_test_table_{short_uid()}"
+
+        template = EHT.load_sfn_template(EHT.AWS_SERVICE_DYNAMODB_PUT_ITEM)
+        definition = json.dumps(template)
+
+        exec_input = json.dumps(
+            {
+                "TableName": table_name,
+                "Item": {"data": {"S": "HelloWorld"}, "id": {"S": "id1"}},
+            }
+        )
+        create_and_record_execution(
+            aws_client.stepfunctions,
+            create_iam_role_for_sfn,
+            create_state_machine,
+            snapshot,
+            definition,
+            exec_input,
+        )
+
+    @markers.snapshot.skip_snapshot_verify(
+        paths=[
+            "$..error"  # TODO: LS returns a ResourceNotFoundException instead of reflecting the validation error
```

Comment:
:+1:  I like the addtional comments, makes it much easier later to figure out why this was skipped